### DISCUSSION TITLE
How to use ForEach control get collection value from customer activity

### DISCUSSION THREAD

**Discussion by futureneo86** [elsa-workflows/elsa-core#295] [Q&A]
Hi, 
I have an custom activity named "ApproveUser"
protected override async Task<ActivityExecutionResult> OnExecuteAsync(WorkflowExecutionContext context, CancellationToken cancellationToken)
        {
 var user = new User
                {                   
                    Name = "LTA"
                    Email =new List<string>(){"email01@gmail.com","email02@gmail.com","email03@gmail.com"},
                    Status=1
                };
                Output.SetVariable("User", user);
return Done();
}
Moreover, I want to user ForEach control to loop Activities.ApproveUser.User.Email, and then sendemail to each element of Email. How to design it?

Thanks for your support

**Comment by sfmskywalker** [comment]
Nice use case!

So first, you would add a `ForEach` activity to the workflow (connecting the User.Done outcome to it).
Then set the `ForEach.CollectionExpression` to either a Liquid or JavaScript expression:

Liquid: `{ Variables.User.Email }`
JavaScript: `(User.Email)` (make sure to include the parentheses; there's a quirk in JINT that would otherwise return an empty value).

Then connect the `ForEach.Iterate` outcome to the `SendEmail` activity, and connect `SendEmail.Done` back to the `ForEach` activity, which closes the loop.

When all emails have been sent, the `ForEach.Done` outcome will be executed, so you can continue the workflow using that outcome (if you have any additional activities to execute).

**Comment by jayachandra21** [comment]
@sfmskywalker 
Even I do have same type of requirement and follow the same approach design the workflow. In ForEach loop i have set ForEach.CollectionExpression as "JavaScript" expression as (User.Email). 
Getting the following error in the ForEach loop: **Error while evaluating JavaScript expression "(User.Email)". Message: User is not defined**
